docs/sphinx_pyodide/sphinx_pyodide/jsdoc.py

```diff
@@ -78,29 +78,70 @@ def get_object_from_json(self, json):
         obj.kind = kind
         return obj
 
+    def flatten_suffix_tree(self):
+        """Flatten self._doclets_by_path into a dictionary self.doclets.
```

Comment:
If possible it would have good nice to have a test (or possibly even a doctest) just for this function